projects/packages/my-jetpack/_inc/components/product-interstitial/jetpack-ai/product-page.jsx

```diff
@@ -0,0 +1,370 @@
+		requestsCount: allTimeRequests = 0,
+		currentTier,
+		nextTier,
+		usagePeriod: usage,
+		isOverLimit,
+	} = aiAssistantFeature || {};
+
+	const hasUnlimited = currentTier?.value === 1;
+	const isFree = currentTier?.value === 0;
+	const hasPaidTier = ! isFree && ! hasUnlimited;
+	const shouldContactUs = ! hasUnlimited && hasPaidTier && ! nextTier && currentTier;
+	const freeRequestsLeft = isFree && 20 - allTimeRequests >= 0 ? 20 - allTimeRequests : 0;
+	const showCurrentUsage = hasPaidTier && ! isFree && usage;
+	const showAllTimeUsage = hasPaidTier || hasUnlimited;
+	const contactHref = getRedirectUrl( 'jetpack-ai-tiers-more-requests-contact' );
+	const feedbackURL = getRedirectUrl( 'jetpack-ai-feedback' );
+
+	// isRegistered works as a flag to know if the page can link to a post creation or not
+	const ctaURL = isRegistered
+		? 'post-new.php?use_ai_block=1&_wpnonce=' + window?.jetpackAi?.nonce
+		: '#/connection';
+	const newPostCta = __( 'Create new post', 'jetpack-my-jetpack' );
+	const installCta = __(
+		'Connect to Jetpack to start using the AI Assistant',
+		'jetpack-my-jetpack'
+	);
+
+	const showRenewalNotice = isOverLimit && hasPaidTier;
+	const showUpgradeNotice = isOverLimit && isFree;
+
+	const currentTierValue = currentTier?.value || 0;
+	const currentUsage = usage?.[ 'requests-count' ] || 0;
+	const tierRequestsLeft =
+		currentTierValue - currentUsage >= 0 ? currentTierValue - currentUsage : 0;
+
+	const renewalNoticeTitle = __(
+		"You've reached your request limit for this month",
+		'jetpack-my-jetpack'
+	);
+	const upgradeNoticeTitle = __( "You've used all your free requests", 'jetpack-my-jetpack' );
+
+	const renewalNoticeBody = sprintf(
+		// translators: %d is the number of days left in the month.
+		__(
+			'Wait for %d days to reset your limit, or upgrade now to a higher tier for additional requests and keep your work moving forward.',
+			'jetpack-my-jetpack'
+		),
+		Math.floor( ( new Date( usage?.[ 'next-start' ] ) - new Date() ) / ( 1000 * 60 * 60 * 24 ) )
+	);
+	const upgradeNoticeBody = __(
+		'Reach for More with Jetpack AI! Upgrade now for additional requests and keep your momentum going.',
+		'jetpack-my-jetpack'
+	);
+	const renewalNoticeCta = sprintf(
+		// translators: %s is the next upgrade value
+		__( 'Get %s requests', 'jetpack-my-jetpack' ),
+		nextTier?.value || 'more'
+	);
+	const upgradeNoticeCta = __( 'Upgrade now', 'jetpack-my-jetpack' );
+
+	const navigateToPricingTable = useMyJetpackNavigate( '/add-jetpack-ai' );
+	const { recordEvent } = useAnalytics();
+
+	const contactClickHandler = useCallback( () => {
+		recordEvent( 'jetpack_ai_upgrade_contact_us', { placement: 'product-page' } );
+	}, [ recordEvent ] );
+
+	const upgradeClickHandler = useCallback( () => {
+		recordEvent( 'jetpack_ai_upgrade_button', {
+			placement: 'product-page',
+			context: 'my-jetpack',
+			current_tier_slug: currentTier?.slug || '',
+			requests_count: allTimeRequests,
+		} );
+		navigateToPricingTable();
```

Comment:
Any reason not to send the user straight to the checkout with the next tier here instead of back to the pricing table?

That would allow you to drop a lot of the complexity around the "current" and "next" tier in the product class.

If the pricing table only shows to user without a purchase, you would only need to worry about the free and 100 tiers showing there.

Reply:
Good point. I'll bring up the subject on our meetup when we'll discuss a bit about this project. The only reason here is to present the user with the next upgrade, something we do from all upgrade flows on the block and logo generator, but it would also make sense to just direct to checkout.

As for the pricing table, that's the target of all the upgrade buttons, so that's why I put some effort there to show the next upgrade vs current. Not saying that's perfect, just what we've been doing so far. I'll keep you posted on what we discuss.

Thanks for taking a look!